### ISSUE TITLE
Not auto-completing (Ubuntu 14.04)

### ISSUE THREAD

**Issue by tpoisot** [apcshields/autocomplete-bibtex#12]
I've tried with the example in the README (and followed the configuration), but I was unable to get any suggestions.

Any ideas about the steps I should take to troubleshoot that?

**Comment by apcshields** [comment]
Timothée,




Let me doublecheck that updates to autocomplete-plus haven't broken anything.




Andrew



—

**Comment by apcshields** [comment]
Timothée,

Have you made sure that autocomplete-plus was successfully installed?

Andrew

**Comment by tpoisot** [comment]
I did. It has been installed, and the path to the bib file is right too.

**Comment by tpoisot** [comment]
Here is the console log:



~~~
Failed to activate package named 'autocomplete-bibtex' TypeError: Cannot read property 'length' of undefined
    at BibtexProvider.module.exports.BibtexProvider.buildWordList (/home/tpoisot/.atom/packages/autocomplete-bibtex/lib/provider.coffee:94:31)
    at BibtexProvider.buildWordList (/home/tpoisot/.atom/packages/autocomplete-bibtex/lib/provider.coffee:3:61)
    at new BibtexProvider (/home/tpoisot/.atom/packages/autocomplete-bibtex/lib/provider.coffee:37:12)
    at Object.module.exports.activate (/home/tpoisot/.atom/packages/autocomplete-bibtex/lib/main.coffee:13:30)
    at Package.module.exports.Package.activateNow (/usr/share/atom/resources/app/src/package.js:235:27)
    at /usr/share/atom/resources/app/src/package.js:221:28
    at Package.module.exports.Package.measure (/usr/share/atom/resources/app/src/package.js:167:15)
    at Package.module.exports.Package.activate (/usr/share/atom/resources/app/src/package.js:215:14)
    at PackageManager.module.exports.PackageManager.activatePackage (/usr/share/atom/resources/app/src/package-manager.js:464:21)
    at /usr/share/atom/resources/app/src/package-manager.js:445:29
~~~

**Comment by apcshields** [comment]
Timothée,

(Sorry—I meant to suggest you send me the log, but I haven’t been at my email much in the last few days.)

Aha! This almost certainly means that it isn’t able to parse your BibTeX file. I can (and should) ask it to suppress the error, but if you are willing to send me the file you’re using, I may be able to get the parser to handle it correctly as well.

Thanks!
Andrew

**Comment by tpoisot** [comment]
Here is an additional log line that I've missed:

~~~

 Error: EISDIR, illegal operation on a directorycode: "EISDIR"errno: -21message: "EISDIR, illegal operation on a directory"stack: (...)get stack: function () { [native code] }arguments: nullcaller: nulllength: 0name: ""prototype: StackTraceGetterconstructor: function () { [native code] }arguments: nullcaller: nulllength: 0name: ""prototype: StackTraceGetterconstructor: function () { [native code] }__proto__: Object__proto__: function Empty() {}<function scope>__proto__: Object__proto__: function Empty() {}<function scope>set stack: function () { [native code] }syscall: "read"__proto__: d/home/tpoisot/.atom/packages/autocomplete-bibtex/lib/provider.coffee:124 module.exports.BibtexProvider.readBibtexFiles
~~~

I've added a gist with some refs here: https://gist.github.com/84f11b0d712741c85967
Thanks!

**Comment by apcshields** [comment]
Timothée,

I get the same error when I use the name of a directory in the ‘Bibtex’ configuration field instead of the name of a file.

So just to make sure, you’re providing the path to a file rather than to a directory, right?

Thanks!
Andrew

**Comment by tpoisot** [comment]
Yes, here is the relevant part of my config

~~~ cson
  "autocomplete-plus":
    fileBlacklist: [
      ".*"
    ]
  "autocomplete-bibtex":
    bibtex: "/home/tpoisot/.pandoc/default.bib"
    fileBlacklist: "!*.{md,markdown,pandoc}"
~~~

**Comment by apcshields** [comment]
Timothée,

I've tweaked a few things and released a new version, 0.5.3. Please update to that version, restart Atom, and let me know if the output in the developer console changes.

Thanks!
Andrew

**Comment by tpoisot** [comment]
Ok, in #14 I propose a fix for #13 and #12 -- confirmed to work on my machine
